File: processor-base/src/main/java/testaccessors/internal/AnnotationVerifier.java
package testaccessors.internal;

import javax.lang.model.element.Element;

abstract class AnnotationVerifier<Annotation extends java.lang.annotation.Annotation> {
  final Lazy<Logger> logger;

  AnnotationVerifier(final Lazy<Logger> logger) {
    this.logger = logger;
  }

  abstract Class<Annotation> annotationClass();

  abstract boolean verify(Element element, Annotation annotation);

  final boolean verify(final Element element) {
    return verify(element, element.getAnnotation(annotationClass()));
  }
}

File: processor-base/src/main/kotlin/testaccessors/internal/base/AnnotationVerifier.kt
package testaccessors.internal.base

import javax.lang.model.element.Element

abstract class AnnotationVerifier<Annotation : kotlin.Annotation>(val logger: Logger) {
  abstract fun annotationClass(): Class<Annotation>

  abstract fun verify(
      element: Element,
      annotation: Annotation = element.getAnnotation<Annotation>(annotationClass())): Boolean
}
